### PR TITLE
Js

### PR DESCRIPTION
**Pull request by scottrogowski**
Add Javascript support, add better testing to Python, and do some cleanup

### CODE REVIEW

File: lib/python.py [hunk 1]
```diff
@@ -234,69 +173,91 @@ def separate_namespaces(tree):
...
+            class_group.add_node(Python.make_nodes(node_tree, parent=class_group)[0])
 
         for subgroup_tree in subgroup_trees:
-            file_group.add_subgroup(_make_class_group(subgroup_tree, parent=file_group))
+            logging.warning("Code2flow does not support nested classes. Skipping %r in %r.",
+                            subgroup_tree.name, parent.token)
+        return class_group
+
+    # @staticmethod
+    # def make_file_group(tree, filename):
```
Comment (suetsuguk) [inline lib/python.py:239]: The `make_file_group()` methods in `python.py` and `javascript.py` are moved to `engine.py`. You can delete this comment block. And the abstract method of the `BaseLanguage` class in `model.py` is not used.

File: lib/javascript.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+            ret += v
+    return ret
+
+
+def get_acorn_version():
+    """
+    Get the version of installed acorn
+    :rtype: str
+    """
+    return subprocess.check_output(['npm', '-v', 'acorn'])
```
Comment (suetsuguk) [inline lib/javascript.py:219]: `npm -v acorn` returns the version of **npm** itself. You can find the version of  the **installed** package with `npm list` or `node -p` as bellow;
```console
$ npm install acorn@8.1.0

$ npm list --depth=0 acorn
/home/foo/code2flow
└── acorn@8.1.0

$ node -p "require('acorn/package.json').version"
8.1.0
```

Comment (scottrogowski) [reply lib/javascript.py]: That was a big oversight on my part. Thank you! Fixed.

File: lib/python.py (hunk 1 above)
Comment (scottrogowski) [reply lib/python.py]: Fixed

File: lib/model.py [hunk 3]
```diff
@@ -23,44 +40,64 @@ def is_installed(executable_cmd):
...
     @abc.abstractmethod
-    def RESERVED_KEYWORDS(self):
-        """
-        :rtype: list[str]
-        """
+    def assert_dependencies():
+        """"""
 
     @staticmethod
     @abc.abstractmethod
```
Comment (suetsuguk) [inline lib/model.py:103]: The abstract `def get_tree(filename)` method should take two arguments `filename` and `source_type`. The `source_type` value is used by the Javascript class.

File: lib/get_ast.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const fs = require('fs');
+const {Parser} = require("acorn")
+
+const sourceType = process.argv[2]
+const src = fs.readFileSync(process.argv[3], 'utf8')
+const tree = Parser.parse(src, {'locations': true, 'sourceType': sourceType,
+                                'ecmaVersion': '2020'})
+const classFields = require('acorn-class-fields');
```
Comment (suetsuguk) [inline lib/get_ast.js:8]:
```suggestion
```

The `acorn-class-fields` package is not used in this script. Can you remove this line?
It causes an error saying "Cannot find module 'acorn-class-fields'" if it is not installed.

Comment (scottrogowski) [reply lib/get_ast.js]: Fixed. Thank you

**Pull request merged**